netket/nn/symmetric_linear.py

```diff
@@ -125,25 +101,23 @@ class DenseSymmMatrix(Module):
 
     def setup(self):
         # pylint: disable=attribute-defined-outside-init
-        perms = np.asarray(self.symmetries)
-        self.n_symm, self.n_sites = perms.shape
-        self.n_hidden = self.features * self.n_symm
-
-        self.symm_cols = jnp.asarray(_symmetrizer_col(perms, self.features))
+        self.n_symm, self.n_sites = np.asarray(self.symmetries).shape
 
     def full_kernel(self, kernel):
         """
-        Converts the symmetry-reduced kernel of shape (n_sites, features) to
-        the full Dense kernel of shape (n_sites, features * n_symm).
+        Converts the symmetry-reduced kernel of shape (out_features, in_features, n_sites) to
+        the full dense kernel of shape (in_features*n_sites, out_features*n_symm).
         """
-        kernel = kernel.transpose(1, 0).reshape(-1)
-        result = kernel[self.symm_cols]
-        return result.reshape(self.n_sites, -1)
+        result = jnp.take(kernel, jnp.asarray(self.symmetries), 2)
+        result = result.transpose(1, 3, 0, 2).reshape(
+            self.n_sites * self.in_features, self.n_symm * self.out_features
+        )
+        return result
```

Comment:
Could we have a comment here that explains the mapping from reduced to full kernel matrix? I remember @PhilipVinc asking me to add a docstring to the now removed `_symmetrizer_col` method and I think it indeed helps people reading the code to have a proper description.

Reply:
have a look at the comments I've just added